src/Compilers/CSharp/Test/Emit2/Semantics/RefReadonlyParameterTests.cs

```diff
@@ -3085,4 +3085,100 @@ .maxstack  2
             }
             """);
     }
+
+    [Fact]
+    public void Invocation_Dynamic()
+    {
+        var source = """
+            class C
+            {
+                void M(ref readonly int p) => System.Console.Write(p);
+                static void Main()
+                {
+                    dynamic d = 1;
+                    var c = new C();
+                    try
+                    {
+                        c.M(d);
+                    }
+                    catch (Microsoft.CSharp.RuntimeBinder.RuntimeBinderException)
+                    {
+                        System.Console.Write("exception");
```

Comment:
nit: in the future, consider adding spaces at the end of such strings (that are meant to be concatenated into the output), I was looking for "exception2" in the source and couldn't find it, haha